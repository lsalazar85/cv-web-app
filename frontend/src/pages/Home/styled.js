import styled from 'styled-components'

const HomeContainer = styled.main`
    display: flex;
    flex-direction: column; 
    align-items: center;
    justify-content: center;
    background: #0a0000;
    color: #888888;
    height: 100vh;
`
const HomeContent = styled.section`
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    border-radius: 5px;
    background: #1d1f21;
    max-height: 100%;
    max-width: 100%;

    @media (min-width: 320px) {width: 100%; height: 100%;}
    @media (min-width: 992px) {width: 90%; height: 90%;}
    @media (min-width: 1100px) {width: 1140px;}    
`

const Side = styled.section`
    display: flex;
    flex-grow: 1;
    
    @media (min-width: 320px) {flex-direction: column; align-items: space-between;}
    @media (min-width: 992px) {flex-direction: row; justify-content: space-between;}
`

export { HomeContainer, HomeContent, Side }