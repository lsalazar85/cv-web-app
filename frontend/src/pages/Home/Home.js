import React from 'react'

import MainContent from '../../components/MainContent/MainContent'
import SideMenu from '../../components/SideMenu/SideMenu'
import BoxName from '../../components/BoxName'
import Footer from '../../components/Footer/Footer'

import { HomeContainer, HomeContent, Side } from './styled'

const Home = () => (
  <HomeContainer>
    <HomeContent>
      <SideMenu />
      <Side>
        <BoxName />
        <MainContent />
      </Side>
    </HomeContent>
    <Footer />
  </HomeContainer>
)

export default Home
