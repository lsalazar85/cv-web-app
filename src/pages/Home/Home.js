import React from 'react';

import MainContent from '../../components/MainContent/MainContent';
import SideMenu from '../../components/SideMenu/SideMenu';
import Person from '../../components/Person';
import Footer from '../../components/Footer/Footer';

import { HomeContainer, HomeContent, Side } from './styled';

const Home = () => (
  <HomeContainer>
    <HomeContent>
      <SideMenu />
      <Side>
        <Person />
        <MainContent />
      </Side>
    </HomeContent>
    <Footer />
  </HomeContainer>
)

export default Home