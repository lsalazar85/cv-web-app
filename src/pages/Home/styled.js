import styled from 'styled-components'

const HomeContainer = styled.div`
    @media (min-width: 320px){
        display: flex;
        flex-direction: column; 
        align-items: center;
        justify-content: center;
        background: #0a0000;
        color: #888888;
        height: 100vh;
    }
`
const HomeContent = styled.div`
    @media (min-width: 320px){
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        border-radius: 5px;
        background: #1d1f21;
        max-height: 100%;
        max-width: 100%;
        width: 100%; 
        height: 100%;
    }
    @media (min-width: 992px){width: 90%; height: 90%;}
    @media (min-width: 1100px){width: 1140px;}    
`

const Side = styled.div`
    @media (min-width: 320px){
        display: flex;
        flex-direction: column;
        flex-shrink: 0;
        flex-grow: 1;
    }
    @media (min-width: 992px){flex-direction: row;}
`

export { HomeContainer, HomeContent, Side }