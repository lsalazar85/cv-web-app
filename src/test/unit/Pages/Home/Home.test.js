import React from 'react';
import { render, cleanup } from '@testing-library/react';

import Home from '../../../../pages/Home';

describe('Home', () => {
    afterEach(cleanup);

    test('should render correctly', () => {
        const { asFragment } = render(
          <Home />
        );
    
        expect(asFragment()).toBeDefined();
        expect(asFragment()).toMatchSnapshot();
    });

})