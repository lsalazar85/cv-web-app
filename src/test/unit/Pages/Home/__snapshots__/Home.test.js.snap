// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx jREUZL"
  >
    <div
      class="sc-gtssRu eFRkeg"
    >
      <div>
        Experience
      </div>
      <div>
        Education
      </div>
      <div>
        Dates
      </div>
    </div>
  </div>
</DocumentFragment>
`;
